developers/advanced/null-value-support.md

```diff
@@ -35,20 +26,90 @@ The following table summarizes the behavior of null handling support in Pinot:
 
 ## Default behavior
 
-Pinot always stores column values in a [forward index](forward-index.md).
-Forward index does never store null values.
-They have to store a value for each row.
+Pinot always stores column values in a [forward index](../../basics/indexing/forward-index.md).
+Forward index never stores null values but have to store a value for each row.
 Therefore independent of the null handling configuration, Pinot always stores a default value for nulls rows in the forward index.
 The default value used in a column can be specified in the [schema](../../configuration-reference/schema.md) 
 configuration by setting the `defaultNullValue` field spec.
+
+{% hint style="info" %}
 Remember that in the JSON used as table configuration, `defaultNullValue` must always be a String.
 If the column type is not String, Pinot will convert that value to the column type automatically.
+{% endhint %}
 
 ## Store nulls at ingestion time
 
-In order to actually store nulls in your segments, you have to enable the `nullHandlingEnabled` configuration in table 
-configuration the [tableIndexConfig.nullHandlingEnabled](../../configuration-reference/table#table-index-config) before 
-ingesting the data.
+In order to support any kind of null handling, the first thing that must be done is to store null values in the segments.
+As said above, independently on whether null storing is enabled or not, the forward index will always store the default
+value for null rows.
+When null storing is enabled, Pinot will create a new index called the _null index_ or _null vector index_.
+This index stores the document ids of the rows that have null values for the column.
+
+{% hint style="danger" %}
+Although storing null handling can be enabled after data has been ingested, data ingested before this mode is enabled
+will not store the null index and therefore it will be treated as not null.
+{% endhint %}
+
+Null storing support is defined per table. 
+This means that one table may be configured to store nulls while another table may not.
+There are two ways to define null storing support in Pinot:
+
+1. [Column based null handling](#column-based-null-handling), where each column inside the table can be configured as 
+nullable or not nullable. 
+This is the recommended way to enable null storing support and only way to support null handling in 
+[multi-stage query engine](../../reference/multi-stage-engine.md).
+2. [Table based null handling](#table-based-null-handling), where all columns in the table are considered nullable.
+This was the only way to handle null values in Pinot before 1.1.0, but it is considered deprecated.
```

Comment:
Because even it is deprecated, it is still supported. Which means they don't _have_ to migrate to the new way in order to store nulls.

Given the new alternative has been recently released, all customers are using right now this deprecated version. It may look strange for them to look for the way they are using right now and not see any information here. But we can move it somewhere else if you think it would be cleaner